### ISSUE TITLE
onnx->trt in C++   get error:this 是 nullptr

### ISSUE THREAD

**Issue by YzbDoMore** [mingj2021/segment-anything-tensorrt#10]
comment the params "orig_im_size"
@torch.no_grad()
    def forward(
        self,
        image_embeddings: torch.Tensor,
        point_coords: torch.Tensor,
        point_labels: torch.Tensor,
        mask_input: torch.Tensor,
        has_mask_input: torch.Tensor,
        # orig_im_size: torch.Tensor,
    ):
        orig_im_size= torch.tensor([1500, 2250], dtype=torch.float)
在C++端加载onnx,在进行onnx->trt时报错
nvinfer.h中return mImpl->getName();
报错:this 是 nullptr。

**Comment by YzbDoMore** [comment]
Package                  Version
------------------------ ------------
absl-py                  1.4.0
certifi                  2022.12.7
charset-normalizer       2.1.1
cmake                    3.25.0
coloredlogs              15.0.1
contourpy                1.1.0
cycler                   0.11.0
filelock                 3.9.0
flatbuffers              23.5.26
fonttools                4.42.1
humanfriendly            10.0
idna                     3.4
importlib-resources      6.0.1
Jinja2                   3.1.2
kiwisolver               1.4.5
lit                      15.0.7
MarkupSafe               2.1.2
matplotlib               3.7.2
mpmath                   1.2.1
networkx                 3.0
numpy                    1.23.4
nvidia-cublas-cu11       11.11.3.6
nvidia-cublas-cu12       12.2.5.6
nvidia-cuda-nvrtc-cu11   11.8.89
nvidia-cuda-nvrtc-cu12   12.2.140
nvidia-cuda-runtime-cu11 11.8.89
nvidia-cuda-runtime-cu12 12.2.140
nvidia-cudnn-cu11        8.9.4.25
nvidia-cudnn-cu12        8.9.4.25
onnx                     1.10.0
onnxruntime              1.14.1
opencv-python            4.7.0.72
packaging                23.1
Pillow                   9.3.0
pip                      23.2.1
prettytable              3.8.0
protobuf                 3.20.3
pycocotools              2.0.7
pyparsing                3.0.9
python-dateutil          2.8.2
pytorch-quantization     2.1.2
PyYAML                   6.0.1
requests                 2.28.1
scipy                    1.10.1
setuptools               68.1.2
six                      1.16.0
sphinx-glpi-theme        0.3
sympy                    1.11.1
tensorrt                 8.5.1.7
tensorrt-bindings        8.6.1
tensorrt-libs            8.6.1
torch                    2.0.1+cu117
torchaudio               2.0.2+cu117
torchvision              0.15.2+cu117
tqdm                     4.66.1
triton                   2.0.0
typing_extensions        4.4.0
urllib3                  1.26.13
wcwidth                  0.2.6
wheel                    0.41.2
zipp                     3.16.2

**Comment by mingj2021** [comment]
orig_im_size 这个参数不需要啊，

**Comment by mingj2021** [comment]
这个参数 跟后处理有关，在模型导出前，我把这部分代码在  forward中删除了，你可以试试或者想一想，为什么要删除。这个工程我不在维护了，好久了，没什么人气，本意也是介绍模型转化的常规思路。

**Comment by YzbDoMore** [comment]
> 这个参数 跟后处理有关，在模型导出前，我把这部分代码在 forward中删除了，你可以试试或者想一想，为什么要删除。这个工程我不在维护了，好久了，没什么人气，本意也是介绍模型转化的常规思路。

能否解答一下为什么要在forward中删除吗，对这块不是很了解，请解惑一下谢谢

**Comment by mingj2021** [comment]
orig_im_size ，跟这个变量相关的代码，涉及到动态分配内存问题，而c++中 trt 需要提前为 input output分配内存的（创建上下文context时）， 当trt 执行到跟这变量这块相关代码时，就会报错的，所以在forward函数中，删除这段代码，并移植到 后处理部分。

**Comment by mingj2021** [comment]
可以自己把这个参数加进去，调试看看，动手实现，预处理 和后处理 这段代码，任何模型都绕不开这块的，建议平时多练练哈！
